frontend_tests/node_tests/server_events.js

```diff
@@ -14,24 +15,665 @@ set_global('window', {
+        user_email: 'bob@example.com',
+        subscriptions: [
+            {
+                stream_id: 42
+                // etc.
+            }
+        ]
+    },
+
+    subscription__update: {
+        type: 'subscription',
+        op: 'update',
+        name: 'devel',
+        property: 'color',
+        value: 'black'
+    },
+
+    update_display_settings__default_language: {
+        type: 'update_display_settings',
+        setting_name: 'default_language',
+        default_language: 'fr'
+    },
+
+    update_display_settings__left_side_userlist: {
+        type: 'update_display_settings',
+        setting_name: 'left_side_userlist',
+        left_side_userlist: true
+    },
+
+    update_display_settings__twenty_four_hour_time: {
+        type: 'update_display_settings',
+        setting_name: 'twenty_four_hour_time',
+        twenty_four_hour_time: true
+    },
+
+    update_global_notifications: {
+        type: 'update_global_notifications',
+        notification_name: 'enable_stream_sounds',
+        setting: true
+    },
+
+    update_message_flags__read: {
+        type: 'update_message_flags',
+        operation: 'add',
+        flag: 'read',
+        messages: [5, 999]
+    },
+
+    update_message_flags__starred: {
+        type: 'update_message_flags',
+        operation: 'add',
+        flag: 'starred',
+        messages: [7, 99]
+    }
+};
+
 var server_events = require('js/server_events.js');
 
+function assert_same(actual, expected) {
+    // This helper prevents us from getting false positives
+    // where actual and expected are both undefined.
+    assert(expected);
+    assert.deepEqual(actual, expected);
+}
+
+// TODO: move this into library
+function capture_args(res, arg_names) {
+    // This function returns a function that, when
+    // arg_names are ['foo', 'bar'] sets res.foo
+    // to the first arg passed in and res.bar to
+    // the second args passed in.  (It's basically
+    // a mock.)
+
+    _.each(res, function (value, key) {
+        delete res[key];
+    });
+
+    return function () {
+        var my_arguments = _.clone(arguments);
+        _.each(arg_names, function (name, i) {
+            res[name] = my_arguments[i];
+        });
+    };
+}
+
+(function test_dispatch_event() {
+    server_events.home_view_loaded();
+    var event;
+
+    // We only ever mock one function at a time,
+    // so we can have a little helper.
+    var args = {}; // for stubs to capture args
+    function capture(names) {
+        return capture_args(args, names);
+    }
+
+    var dispatch = function (ev) {
+        server_events._get_events_success([ev]);
+    };
+
+    // alert_words
+    event = event_fixtures.alert_words;
+    add_dependencies({alert_words: 'js/alert_words.js'});
+    dispatch(event);
+    assert_same(global.alert_words.words, ['fire', 'lunch']);
+
+    // default_streams
+    event = event_fixtures.default_streams;
+    set_global('admin', {update_default_streams_table: noop});
```

Comment:
Given how much this file makes use of `set_global` to overwrite things, perhaps it makes sense to at the end of each mini-test do something to restore the state of globals?  I worry that this current implementation could end up with some very difficult to debug problems in the future where a future test is impacted by these `set_globals` for earlier tests...